File: runtimejs/java/lang/Iterable.js
//load// java/lang/Object
var java_lang_Iterable = 
{   _superinterfaces: [], 
    _defaults:
    {   forEach_1: function(consumer)
        {   var i = this.iterator_0();
            while (i.hasNext_0()) 
            {   consumer.accept_1(i.next_0());
            }
        }
        ,
    }
}; 

// methods:
// Iterator<T> iterator()
// default void forEach(Consumer<T> consumer)

File: runtimecs/java/lang/Iterable.cs
namespace java.lang 
{
    public interface Iterable
    {
        java.util.Iterator iterator();
        void forEach(java.util.function.Consumer consumer);
    }

    public static class Iterable_0009
    {
        public static void forEach(Iterable @this, java.util.function.Consumer consumer)
        {   
            java.util.Iterator i = @this.iterator();
            while (i.hasNext()) { consumer.accept(i.next()); }
        }
    }
}
